python/shark_turbine/dynamo/tensor.py

```diff
@@ -372,6 +466,132 @@ def _rand(*args, dtype=None):
     return t
 
 
+@functools.lru_cache(maxsize=None)
+def _get_device_state() -> DeviceState:
+    return DeviceState(driver="local-task")
+
+
+# Inspiration from https://github.com/nod-ai/SHARK-Turbine/blob/8293de5414889c72ff5cd10bf33c43fb0a3ea3ee/python/shark_turbine/aot/builtins/jittable.py#L212-L237
+# and https://github.com/nod-ai/SHARK-Turbine/blob/main/python/shark_turbine/dynamo/backends/cpu.py
+# TODO: Try to generalize for other devices.
+@compute_factory(torch.add)
```

Comment:
May want to generalize compute_factory to take posargs or a tuple of match predicates so it can be checked with a fast `in`. As written, this will do five extra function calls.

Reply:
I am not sure I fully understood this one. 😅 
But if IIUC, you are recommendding we just get rid of `@compute_factory` and have `self.COMPUTE_METHODS` used in `__torch_function__` be pre-initialized/filled? 🤔

Reply:
I may have read this wrong. Was assuming that @compute_factory was creating new function wrappers at each level vs mutating side state.